File: backend/models/roles.go
package models

import "gorm.io/gorm"

// アプリケーションのデータモデルを表現し、データベースのテーブルを表現するための構造体
type Roles struct {
	gorm.Model              // ID unit を含む構造体となっている
	RoleID     uint         `gorm:"not null;unique"`
	RoleName   string       `gorm:"not null"`
	UsersRoles []UsersRoles `gorm:"foreignKey:RoleID;references:RoleID;constraint:OnDelete:CASCADE"` // UsersRolesへの値は指定しない。GORMがリレーションを自動的に処理
}

File: frontend/src/data/Roles.ts
// Roles.ts
export interface Role {
    id: number;
    role_name: string;
  }
  
  export const roles: Role[] = [
    {
      id: 1,
      role_name: "管理者",
    },
    {
      id: 2,
      role_name: "一般ユーザー",
    },
  ];
  